examples/SLAM/js/hit-test-location-root.ts

```diff
@@ -0,0 +1,102 @@
+import { Component, Object as WLEObject, Type } from '@wonderlandengine/api';
+import { ARSession, WebXRProvider, ARProvider } from '../../..';
+/**
+ * Sets up a [WebXR Device API "Hit Test"](https://immersive-web.github.io/hit-test/)
+ * and places the object to the hit location.
+ * 
+ * Difference from the original 'hit-test-location' is that the reticle does not have to be a child of the camera.
+ *
+ * **Requirements:**
+ *  - Specify `'hit-test'` in the required or optional features on the AR button in your html file.
+ */
+
+class HitTestLocationRoot extends Component {
+  public static TypeName = 'hit-test-location-root';
+  public static Properties = {
+    camera: { type: Type.Object },
+  };
+
+  // injected by WLE
+  camera!: WLEObject;
+
+  private tempScaling = new Float32Array(3);
+  private visible = false;
+
+  private xrViewerSpace?: XRReferenceSpace;
+  private xrHitTestSource: XRHitTestSource | null = null;
+
+  private tracking = false;
+  init() {
+    ARSession.onSessionStarted.push(this.onSessionStarted);
+    ARSession.onSessionEnded.push(this.onSessionEnded);
+    this.tempScaling.set(this.object.scalingLocal);
+    this.visible = false;
+    this.object.scale([0, 0, 0]);
+  }
+
+  update() {
+    if (this.tracking) {
+      const wasVisible = this.visible;
+      if (this.xrHitTestSource) {
+
+        const frame = WL.xrFrame;
+        if (!frame)
+          return;
+
+        let hitTestResults = frame.getHitTestResults(this.xrHitTestSource);
+        if (hitTestResults.length > 0) {
+          let pose = hitTestResults[0].getPose(this.xrViewerSpace);
+          this.visible = true;
+
+          // this is good;
+          const tw = this.camera.transformPointWorld(new Array(3), new Array(pose.transform.position.x, pose.transform.position.y, pose.transform.position.z));
+          this.object.setTranslationWorld(tw);
+
+
+          // TODO: how do I get the world ROTATION
```

Comment:
This TODO is up for discussion, I guess the solution would be one of those `toWorldSpaceTransform(pose.transform /* converted to array */)`, but that solution is waiting for the bugfix, if I understand correctly?